### PR TITLE
refactor: speed up partition sort key syncing

### PR DESCRIPTION
**Pull request by crepererum**
Prior to syncing, all chunks have a "locally correct" partiton sort key, i.e. one that at least covers all chunk columns (this is ensured during chunk creation, both for parquet chunks as well as ingester chunks). However due to the timing, some chunks may have a newer (= longer) partition sort key. All we need to do to fix this is to pick the longest partition sort key, there is no need to go through the whole cache system again.

For #6358.

### CODE REVIEW

**Review by alamb — APPROVED** [review]
Makes sense to me

File: querier/src/table/state_reconciler.rs [hunk 1]
```diff
@@ -231,52 +223,34 @@ impl Reconciler {
...
         // write partition sort keys to chunks
         chunks
             .into_iter()
             .map(|chunk| {
                 let partition_id = chunk.partition_id();
-                let sort_key = sort_keys
-                    .get(&partition_id)
-                    .expect("sort key for this partition should be fetched by now");
-                chunk.update_partition_sort_key(sort_key.clone())
+                let sort_key = sort_keys.get(&partition_id);
```
[the remark this answers is not in the document]
Comment (crepererum) [reply querier/src/table/state_reconciler.rs:252]: Nope, for partitions that only have ingester data that is NOT persisted yet, there won't be a sort key.

File: querier/src/table/state_reconciler.rs [hunk 2]
```diff
@@ -231,52 +223,34 @@ impl Reconciler {
...
+    fn sync_partition_sort_keys(
         &self,
         chunks: Vec<Box<dyn UpdatableQuerierChunk>>,
-        span: Option<Span>,
     ) -> Vec<Box<dyn UpdatableQuerierChunk>> {
-        let span_recorder = SpanRecorder::new(span);
-
-        // collect columns
-        let mut all_columns: HashMap<PartitionId, HashSet<ColumnId>> = HashMap::new();
+        // collect latest (= longest) sort key
```
[the remark this answers is not in the document]
Comment (crepererum) [reply querier/src/table/state_reconciler.rs:230]: it's technically not a `prefix`. Sort key can be extended in the middle (but the existing column keep their relative order). I'll adjust that.

File: querier/src/chunk/mod.rs [hunk 3]
```diff
@@ -140,6 +140,11 @@ impl QuerierChunk {
         self.meta.as_ref()
     }
 
+    /// [`Arc`]ed version of the partition sort key.
```
[the remark this answers is not in the document]
Comment (crepererum) [reply querier/src/chunk/mod.rs:143]: done

**Pull request merged**